app/controllers/spree/api/v2/storefront/stripe/payment_intents_controller.rb

```diff
@@ -38,12 +38,37 @@ def update
               spree_authorize! :update, spree_current_order, order_token
 
               @payment_intent = spree_current_order.payment_intents.find(params[:id])
-              @payment_intent.attributes = permitted_attributes
+              @payment_intent.attributes = permitted_attributes.except(:off_session)
               @payment_intent.save!
 
               render_serialized_payload { serialize_resource(@payment_intent) }
             end
 
+            def confirm
+              @payment_intent = SpreeStripe::PaymentIntent.find(params[:id])
+
+              stripe_payment_intent = @payment_intent.stripe_payment_intent
+              order = @payment_intent.order
+
+              if order.canceled?
+                render_error_payload(Spree.t('order_canceled'))
```

Comment:
Normally, `ensure_order` will return `404` when the order is already  `canceled` or `completed`, since we are only searching for incomplete orders in `spree_current_order`.

I've added those custom message to not confuse the user why they are receiving `404` when the payment intent ID is valid

Reply:
@Vegann Maybe we should first identify if the request is authorized? We could compare the payment intent order token to what's being sent in the header and if it belongs to the current user. We can even add a specialized `FindCurrent` service to `spree_stripe` - but maybe that's an overkill.

Reply:
We can't compare the token from the header and the one from the order, since you don't have to send an order token when the user is logged in.

I've added this line `elsif order != spree_current_order` to make sure that the order from the `FindCurrent` is actually the same as in the payment intent

Reply:
I'm not sure if that's a big security risk, but it's possible here to find the payment intent that doesn't belong to the user requesting it. We won't do much with it though, at most we'll read the order status. WDYT?

Reply:
I also don't think is a security risk, since we do not return neither the payment intent nor the order before we confirm that the order belongs to this user/order token.

At most, the user will learn that the order for this payment intent is either canceled or confirmed, but I don't think this is a big deal